Mathlib/Data/Int/Cast/Basic.lean

```diff
@@ -7,59 +7,149 @@ import Mathlib.Data.Int.Cast.Defs
 import Mathlib.Algebra.Group.Basic
 
 /-!
-# Cast of integers
+# Cast of integers (additional theorems)
 
-This file defines the *canonical* homomorphism from the integers into an
-additive group with a one (typically a `ring`).  In additive groups with a one
-element, there exists a unique such homomorphism and we store it in the
-`int_cast : ℤ → R` field.
+This file proves additional properties about the *canonical* homomorphism from
+the integers into an additive group with a one (`int.cast`).
 
-Preferentially, the homomorphism is written as a coercion.
+There is also `Data.Int.Cast.Lemmas`,
+which includes lemmas stated in terms of algebraic homomorphisms,
+and results involving the order structure of `ℤ`.
 
-## Main declarations
-
-* `int.cast`: Canonical homomorphism `ℤ → R`.
-* `add_group_with_one`: Type class for `int.cast`.
+By contrast, this file's only import beyond `Data.Int.Cast.Defs` is `Algebra.Group.Basic`.
 -/
 
+
+universe u
+
 namespace Nat
-variable [AddGroupWithOne R]
+
+variable {R : Type u} [AddGroupWithOne R]
 
 @[simp, norm_cast]
 theorem cast_sub {m n} (h : m ≤ n) : ((n - m : ℕ) : R) = n - m :=
   eq_sub_of_add_eq <| by rw [← cast_add, Nat.sub_add_cancel h]
+#align nat.cast_sub Nat.cast_subₓ
+-- I think this was a dubious translation because of `HasLiftT` appearing in the type signature
+
+@[simp, norm_cast]
+theorem cast_pred : ∀ {n}, 0 < n → ((n - 1 : ℕ) : R) = n - 1
+  | 0, h => by cases h
+  | n + 1, _ => by rw [cast_succ, add_sub_cancel]; rfl
+#align nat.cast_pred Nat.cast_pred
 
 end Nat
 
+open Nat
+
 namespace Int
-variable [AddGroupWithOne R]
+
+variable {R : Type u} [AddGroupWithOne R]
+
+@[simp, norm_cast]
+theorem cast_negSucc (n : ℕ) : (-[n+1] : R) = -(n + 1 : ℕ) :=
+  AddGroupWithOne.intCast_negSucc n
+#align int.cast_neg_succ_of_nat Int.cast_negSuccₓ
```

Comment:
```suggestion
#align int.cast_neg_succ_of_nat Int.cast_negSucc
```
We definitely want to translate these without the ₓ.  The only reason this statement is different is because `coe` is gone now, but this lemma is the closest approximation we've got in Lean 4.  I think the statement is even defeq to the Lean 3 version.

(Note sure about the precise semantics of `#align foo Fooₓ` so maybe the ₓ does what I want.)

Reply:
There is no `ₓ` that appears in the final translation. The `ₓ` is just a signal to mathport that the types don't match and it should use the type of `Foo` downstream instead of `foo`.

I think in this case the dubious translations are arising mostly (if not entirely) from the fact that the nightly on which this mathport output was based came from a mathlib4 before I had ported `Data.{Nat, Int}.Cast.Defs`. I will check again when the next nightly comes out to see if these are still marked as dubious.